src/session.ts

```diff
@@ -104,55 +107,66 @@ export class SessionManager implements Middleware {
 
     public start() {
         this.sessionSettings = Settings.load();
+
         this.log.startNewLog(this.sessionSettings.developer.editorServicesLogLevel);
 
+        // Create the PowerShell executable finder now
+        this.powershellExeFinder = new PowerShellExeFinder(
+            this.platformDetails,
+            this.sessionSettings.powerShellAdditionalExePaths);
+
         this.focusConsoleOnExecute = this.sessionSettings.integratedConsole.focusConsoleOnExecute;
 
         this.createStatusBarItem();
 
-        this.powerShellExePath = this.getPowerShellExePath();
+        try {
+            this.powerShellExePath = this.getPowerShellExePath();
+        } catch (e) {
+            vscode.window.showErrorMessage(
+                "Unable to find PowerShell. Do you have PowerShell installed? See logs for more details.");
```

Comment:
Isn't there an aka.ms URL for taking folks to the download page for various PS releases?  Might be nice to put that URL here.

Reply:
Isn't there an aka.ms URL for taking folks to the download page for various PS releases?  Might be nice to put that URL here.

Reply:
Which aka.ms URL are you thinking of @rkeithhill?

Reply:
Well there is `http://aka.ms/getpowershell` but that redirects to `https://github.com/PowerShell/PowerShell` which is OK but it would be better if it redirected to `https://github.com/PowerShell/PowerShell#get-powershell`.

Reply:
Yeah agreed re the link

Reply:
Ok turns out we don't own that one. Going to put in `https://aka.ms/get-powershell`

Reply:
<img width="470" alt="Screen Shot 2019-10-24 at 10 12 37" src="https://user-images.githubusercontent.com/7009879/67508991-e9132380-f646-11e9-8715-bef64661ab17.png">

So I've confirmed that this happens when there's no PowerShell on a system (kind of tricky since the extension requires PowerShell to build/debug).

So:

- Is the wording ok?
- Is there a way to make the link appear as a link?
- This doesn't prompt for restart (that only happens on server process exit). Is that the right behaviour?
- Anything else I should add?

Reply:
Yeah, that one is better.

Reply:
@rjmholt : I get this error message also when PowerShell Core 7.0 **IS** on the system but was not installed using installer e.g. daily compiled builds unpacked to an folder from which PowerShell 7.0 is started. I guess there should be a check if *pswh.exe* can be found in the *$env:Path*

Reply:
We don't check the path by design. There's no way for us to efficiently know *which* PowerShell that is.

Instead, you want to add that the the `additionalPowerShellExePaths` configuration

Reply:
Ah is that error from strict mode as set in your profile? Hopefully I can fix that.

Reply:
@rjmholt : Yep, I do set *StrictMode* in the profile and thanks for the tip with `additionalPowerShellExePaths`.

Reply:
![Screenshot from 2019-10-24 22-21-34](https://user-images.githubusercontent.com/7009879/67545136-e3502900-f6ac-11e9-97a9-5f5047be3c91.png)

Ok how about this?

Button takes you to https://aka.ms/get-powershell-vscode. Gave it a nice name, since Ubuntu displays the link to you before opening it.

Reply:
Awesome!

Reply:
>  I do specify PowerShell path in the workspace config

I think that one's for VSCode's default shell, rather than what the PowerShell extension uses.

Picking through the client code, it looks like we could possibly clean some of that up at some point. But we're pretty focused on the back end at the moment.